src/Sulu/Bundle/MarkupBundle/Markup/HtmlTagExtractor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MarkupBundle\Markup;
+
+/**
+ * Extracts tags from html content.
+ */
+class HtmlTagExtractor implements TagExtractorInterface
+{
+    const COUNT_REGEX = '/<%1$s:[a-z]+/';
+    const ATTRIBUTE_REGEX = '/(?<name>\b[\w-]+\b)\s*=\s*"(?<value>[^"]*)"/';
+    const CONTENT_REGEX = '/(?:>(?<content>[^<]*)<)/';
+    const TAG_REGEX = '/(?<tag><%1$s:(?<name>[a-z]+)((?!>|\/>).)*(?:\/>|>((?!<\/%1$s:\2*>).)*<\/%1$s:\2*>))/';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count($html, $namespace)
+    {
+         return preg_match_all(sprintf(self::COUNT_REGEX, $namespace), $html, $matches);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($html, $namespace)
+    {
+        if (!preg_match_all(sprintf(self::TAG_REGEX, $namespace), $html, $matches)) {
+            return [];
+        }
+
+        $sortedTags = [];
+        for ($i = 0, $length = count($matches['name']); $i < $length; ++$i) {
+            $tag = $matches['tag'][$i];
+            $name = $matches['name'][$i];
+            if (!array_key_exists($name, $sortedTags)) {
+                $sortedTags[$name] = [];
+            }
+
+            $sortedTags[$name][$tag] = $this->getAttributes($tag);
+        }
+
+        return $sortedTags;
+    }
+
+    /**
+     * Returns attributes of given html-tag.
+     *
+     * @param string $tag
+     *
+     * @return array
+     */
+    private function getAttributes($tag)
```

Comment:
I'm not really a fan of having such a bunch of private methods in a product which should be easily extensible.
If I've to extend the `HtmlExtractor` in some case and just want to add e.g. one static attribute I've to "copy over" the whole method for just doing `$attributes['some-fancy-attr'] = true;` (whatever the case could be)